src/CakeSlayer/Password.hs

```diff
@@ -0,0 +1,74 @@
+-- | User passwords. Uses @bcrypt@ password hashing.
+
+module CakeSlayer.Password
+       ( PasswordHash (unPasswordHash)
+       , PasswordPlainText (..)
+       , unsafePwdHash
+       , mkPasswordHashWithPolicy
+       , mkPasswordHash
+       , verifyPassword
+       , generateRandomPassword
+       ) where
+
+import Data.Aeson (FromJSON, ToJSON)
+import Database.PostgreSQL.Simple.FromField (FromField)
+import Database.PostgreSQL.Simple.ToField (ToField)
+import Elm (Elm)
+
+import CakeSlayer.Error (WithError, throwOnNothingM)
+import CakeSlayer.Random (mkRandomString)
+
+import qualified Crypto.BCrypt as BC
+
+
+-- | Password hash.
+newtype PasswordHash = PasswordHash
+    { unPasswordHash :: Text
+    } deriving stock (Show, Generic)
+      deriving newtype (Eq, FromField, ToField, FromJSON, ToJSON, Elm)
+
+-- | Unsafe function for constructing 'PasswordHash'. Used mostly for testing.
+unsafePwdHash :: Text -> PasswordHash
+unsafePwdHash = PasswordHash
+
+-- | Password in plain text.
+newtype PasswordPlainText = PasswordPlainText
+    { unPasswordPlainText :: Text
+    } deriving stock (Show, Generic)
+      deriving newtype (Eq, FromJSON, ToJSON, Elm)
+
+
+-- | Generates a password hash given the hashing policy and its plane text.
+-- This has to be done in IO asy generating the salt requires RNG.
```

Comment:
I think we can write evenbetter documentation here:
Mention explicitly that fast hashing policy should be used for tests and slow for production (with the link to `mkPasswordHash`.

Also, what do you think if
```suggestion
-- This has to be done in IO as generating the salt requires RNG.
```